### PR TITLE
finished valid parens

### PR DESCRIPTION
**Pull request by EtiWitty**

### CODE REVIEW

File: algorithm/valid parentheses/validParen.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+		")": "(",
+		"}": "{",
+		"]": "["
+	}
+
+	let stack = [];
+	for (let i = 0; i < str.length; i++) {
+		if (str[i] === "(" || str[i] === "{" || str[i] === "[") {
+			stack.push(str[i]);
+		} else if ( stack[stack.length - 1] === map[str[i]]) {
```
Comment (eyedean) [inline algorithm/valid parentheses/validParen.js]: You might want to check if it doesn't go out of bound of the array. i.e. check if `stack.length - 1` is not going be `-1`.

An example would be the `str` string being `"}()"`.  While your code might work, it's definitely accessing `stack[-1]` in this line, which is not a good practice.

File: algorithm/valid parentheses/validParen.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+			stack.pop();
+		} else {
+			return false;
+		}
+	}
+	return stack.length === 0 ? true : false;
```
Comment (eyedean) [inline algorithm/valid parentheses/validParen.js]: This line can be simplified as `return stack.length === 0;` or even `return !stack.length;`.

File: algorithm/valid parentheses/validParen.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+var isValid = function(str) {
+	let map = {
+		")": "(",
+		"}": "{",
+		"]": "["
+	}
```
Comment (eyedean) [inline algorithm/valid parentheses/validParen.js]: nit - missed semicolon.

File: algorithm/valid parentheses/validParen.js (hunk 2 above)
Comment (EtiWitty) [reply algorithm/valid parentheses/validParen.js]: Gotcha!

File: algorithm/valid parentheses/validParen.js (hunk 3 above)
Comment (EtiWitty) [reply algorithm/valid parentheses/validParen.js]: Done!

**Pull request merged**